src/cls/2pc_queue/cls_2pc_queue_client.cc

```diff
@@ -44,12 +59,31 @@ int cls_2pc_queue_get_capacity(IoCtx& io_ctx, const std::string& queue_name, uin
 #endif
 
 // optionally async method for getting capacity (bytes) 
-// after answer is received, call cls_2pc_queue_get_capacity_result() to prase the results
+// after answer is received, call cls_2pc_queue_get_capacity_result() to parse the results
 void cls_2pc_queue_get_capacity(ObjectReadOperation& op, bufferlist* obl, int* prval) {
   bufferlist in;
   op.exec(TPC_QUEUE_CLASS, TPC_QUEUE_GET_CAPACITY, in, obl, prval);
 }
 
+#ifndef CLS_CLIENT_HIDE_IOCTX
+int cls_2pc_queue_get_entries_and_size(IoCtx& io_ctx, const std::string& queue_name, uint32_t& committed_entries, uint64_t& size) {
+  bufferlist in, out;
+  const auto r = io_ctx.exec(queue_name, TPC_QUEUE_CLASS, TPC_QUEUE_GET_ENTRIES_AND_SIZE, in, out);
+  if (r < 0 ) {
+    return r;
+  }
+
+  return cls_2pc_queue_get_entries_and_size_result(out, committed_entries, size);
```

Comment:
in the other calls, getting the bufferlist and parsing it is doen in 2 different functions

Reply:
I see that there are 2 type of calls to resolve this, one async with **ObjectReadOperation** and the other with **IoCtx**:
```
int cls_2pc_queue_get_topic_stats(IoCtx& io_ctx, const std::string& queue_name, uint32_t& committed_entries, uint64_t& size)

void cls_2pc_queue_get_topic_stats(ObjectReadOperation& op, bufferlist* obl, int* prval) {
```
same thing is done for capacity:
```
int cls_2pc_queue_get_capacity(IoCtx& io_ctx, const std::string& queue_name, uint64_t& size) {

void cls_2pc_queue_get_capacity(ObjectReadOperation& op, bufferlist* obl, int* prval) {
```
I used the context because it is simpler